### PR TITLE
PCH: Redesign the PCH Editor Sidebar

### LINKED ISSUE #2206: Improve the Content Helper's UI
## Is your feature request related to a problem?
The Content Helper could improve UX with a redesign of the UI.

## Describe the solution you'd like
Prototype the redesign and apply it to the Content Helper.

### PR DESCRIPTION
**Pull request by acicovic**
## Description
This PR consolidates all the work that has been done in the context of redesigning the Content Helper Editor Sidebar. This was a big amount of work touching many parts of our codebase, so it was made step by step in smaller PRs which are referenced in this PR's comments.

### Side effects
- As part of this work, parts of our settings system were rewritten. Due to this, some settings in the `Related Posts` and `Performance Details` sections will be reset. Crafting a migration path for this would have taken time without important benefits.

## Motivation and context
Closes #2206.

## How has this been tested?
- Manual testing and QA reviews.
- Automated tests have been updated to pass.

## Screenshots

screenshot-3
screenshot-4
screenshot-5
screenshot-6
screenshot-7

### CODE REVIEW

File: src/content-helper/editor-sidebar/performance-details/performance-details.scss [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+
+		.components-select-control {
+			display: flex;
+			align-items: flex-start;
+			gap: to_rem(16px);
+			align-self: stretch;
+		}
+
+		.components-input-control__prefix {
+			margin-left: to_rem(8px);
```
Comment (dabowman) [inline src/content-helper/editor-sidebar/performance-details/performance-details.scss]: @vaurdan @acicovic I up-streamed this fix to core but they do provide a wrapper for prefixes and suffixes to apply the correct padding. It's mentioned in the storybook entry for inputControl, which is apparently what the selectControl is based on.

File: src/content-helper/editor-sidebar/performance-details/component.tsx [hunk 2]
```diff
@@ -73,233 +232,79 @@ export function PerformanceDetails(
...
+		<div className="wp-parsely-performance-panel">
+			<PerformanceStatPanel
+				title={ __( 'Performance Stats', 'wp-parsely' ) }
+				icon={ moreVertical }
+				dropdownChildren={ ( { onClose } ) => <PerformanceStatsMenu onClose={ onClose } /> }
+			>
+				<div className="panel-settings">
+					<SelectControl
+						value={ settings.PerformanceStatsSettings.Period }
+						prefix={ __( 'Period: ', 'wp-parsely' ) }
```
Comment (dabowman) [inline src/content-helper/editor-sidebar/performance-details/component.tsx]: @acicovic @vaurdan they provide a <InputControlPrefixWrapper> component to wrap the prefix and <InputControlSuffixWrapper> to wrap the suffix. That adds the correct padding when the component has a prefix or suffix added. https://github.com/WordPress/gutenberg/pull/59719 I opened a PR but we can probably implement it the same way here.

Comment (dabowman) [reply src/content-helper/editor-sidebar/performance-details/component.tsx]: InputControlPrefixWrapper and InputControlSuffixWrapper are the ones you want.

Comment (dabowman) [reply src/content-helper/editor-sidebar/performance-details/component.tsx]: There also seem to be a bunch of size options that don't exist in Figma. We want the one that's 40px hight. I think it's called __unstable-large

Comment (vaurdan) [reply src/content-helper/editor-sidebar/performance-details/component.tsx]: Thanks for looking! I will try to implement the prefix and sufix components and will open a PR once that's done.

Comment (vaurdan) [reply src/content-helper/editor-sidebar/performance-details/component.tsx]: Addressed in https://github.com/Parsely/wp-parsely/pull/2278

**Pull request merged**